modules/kubernetes/grafana-agent/templates/kube-state-metrics-values.yaml.tpl

```diff
@@ -0,0 +1,44 @@
+priorityClassName: "platform-low"
+podSecurityPolicy:
+  enabled: false
+metricLabelsAllowlist:
+  - "namespaces=[xkf.xenit.io/kind]"
+namespaces: ${tenant_namespaces_csv}
```

Comment:
@phillebaba  could this be possible to do but with a serviceMonitor.
Only gather metrics from the kube-state-metrics endpoints with the namespace label that matches: X?

Reply:
Yes we had a discussion about this and that may be a better solution for the future. Having the service monitor filter the namespace label. But I think we should skip this for now as I am not sure what would happen to cluster wide resources.

Reply:
Is there any point in kube state metrics scraping more than needed?

Reply:
This limits this kube-state-metrics instance to only look at a specific namespace. Why have it look at all of them and then use the serviceMonitor to filter specific labels? This way is both easier and consumes less resources.

Reply:
The idea is more that one could use a shared KSM. Either way it should not be implemented now.